File: 07_02_Greedy/93_3_Rabbits_in_Forest/szzng.js
var numRabbits = function(answers) {
    let result = 0
    const cnt = {}
    answers.forEach(v=>cnt[v] = v in cnt? cnt[v] + 1 : 1)
    Object.entries(cnt).forEach(([k,v])=>{
        k = parseInt(k)
        result+=Math.ceil(v/(k+1))*(k+1)
    })
    return result
};

File: 07_02_Greedy/93_3_Rabbits_in_Forest/szzng.py
from math import ceil
from typing import List


class Solution:
    def numRabbits(self, answers: List[int]) -> int:
        result = 0
        cnt = {}
        for i in answers:
            cnt[i] = cnt[i] + 1 if i in cnt else 1

        for k, v in cnt.items():
            result += ceil(v / (k + 1)) * (k + 1)

        return result
